pkgs/build-support/nix-gitignore/default.nix

```diff
@@ -41,6 +41,9 @@ in rec {
         let split = match "^(!?)(.*)" l;
         in [(elemAt split 1) (head split == "!")];
```

Comment:
We should probably handle the `\!` to avoid negation at the same time.

Reply:
I pushed a version that handles `!`. I checked that it does not regress the negation logic. (well on 1 example)